### PR TITLE
wip: golang client tut

### PR DESCRIPTION
**Pull request by distractedm1nd**
## Summary by CodeRabbit

- **New Features**
	- Added new functions in the client library to enhance interactions with RPC endpoints: `SubmitBlob`, `SubscribeHeaders`, and `GetEDS`.

- **Documentation**
	- Updated the Golang client tutorial to include guidance on new client functionalities.
	- Revised the Node tutorial to focus more on troubleshooting, removing outdated Golang guide content.

- **Bug Fixes**
	- Ensured consistency in function declarations within the documentation.

### CODE REVIEW

**Comment by distractedm1nd** [comment]
Relevant: 
https://github.com/celestiaorg/celestia-node/issues/3292#issuecomment-2090957367

File: developers/node-tutorial.md [hunk 1]
```diff
@@ -425,6 +425,157 @@ when you created the wallet.
 
 With your wallet funded, you can move on to the next step.
 
+## Golang Client Library Guide {#golang-client-library}
```
Comment (jcstein) [inline developers/node-tutorial.md:428]: let's move this to `developers/golang-client-tutorial.md` or some similar title. i'd do it myself but want you to keep the credit 🫡

File: developers/node-tutorial.md [hunk 2]
```diff
@@ -425,6 +425,157 @@ when you created the wallet.
...
+
+	// Fetch the EDS
+	return client.Share.GetEDS(ctx, header)
+}
+```
+
+### Balances and Transfers
+
+TODO
+
```
Comment (jcstein) [inline developers/node-tutorial.md:578]: until here, could be moved into new doc. what do you think?

then I will tidy up / cut down the existing CLI tutorial

File: developers/submit-data.md [hunk 3]
```diff
@@ -170,6 +170,59 @@ celestia blob submit <hex-encoded namespace> <hex-encoded data>
 
 Learn more in the [node tutorial](./node-tutorial.md).
 
+### The celestia-node API client
```
Comment (jcstein) [inline developers/submit-data.md:173]: lgtm

File: developers/node-tutorial.md [hunk 4]
```diff
@@ -1141,13 +1292,6 @@ The example transaction can be
 If you'd like to create your own SVG, post it to Celestia, and retrieve it,
 you can check out the [Base64 SVG Tutorial](https://based64.xyz/).
 
-### Golang guide
```
Comment (jcstein) [inline developers/node-tutorial.md:1144]: 👍

File: developers/prompt-scavenger.md [hunk 5]
```diff
@@ -1,8 +1,8 @@
 ---
 description: Learn how to interact with the Celestia Node API with this tutorial.
 prev:
-  text: "Node RPC CLI tutorial"
-  link: "/developers/node-tutorial"
+  text: "Golang client tutorial"
+  link: "/developers/golang-client-tutorial"
```
Comment (jcstein) [inline developers/prompt-scavenger.md:5]: the `prev` and `next` frontmatter allow you to customize the text/link for previous and next pages that are shown at the footer. this is something i mostly use here to skip external links in the menu, preserving the nav flow ([page on it in vitepress docs](https://vitepress.dev/reference/default-theme-prev-next-links#prev-next-links))

**Review by jcstein — APPROVED** [review]

**Pull request merged**